bindings/cs/rl.net.cli.test/CleanupContainer.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Rl.Net.Cli.Test
+{
+    public sealed class CleanupContainer : IDisposable
+    {
+        private Stack<IDisposable> cleanupStack = new Stack<IDisposable>();
+
+        public void Dispose()
+        {
+            Stack<IDisposable> localCleanupStack = Interlocked.Exchange(ref cleanupStack, new Stack<IDisposable>());
```

Comment:
Is the idea here that when Dispose is getting called, it's possible some other thread is adding to  cleanupStack?  Is that possible?  i.e. does calling Dispose() imply that there are no reachable instances?  If that is not the case, will the additional items get cleaned up properly since dispose has been already called?

Reply:
This is primarily intended as a best-efforts cleanup mechanism for tests. The exchange is there to ensure that if tests are running in an unexpected multithread situations, we do not start throwing out of cleanup with iteration at the same time as collection changed.

Mostly this is a habit developed for dealing with IDisposables in multi-threaded situations: When it doubt, use Interlocked to null out / reset the shared object, and then clean up locally. The "right" solution to this is to have a separate background queue for calling .Dispose(), but it is overkill for unit tests.